scripts/xtp_autogen_mapping.in

```diff
@@ -75,10 +75,9 @@ def search_cores() -> int:
         return multiprocessing.cpu_count()
 
     with open(cpuinfo, 'r') as handler:
-        for line in handler:
-            if "cpu cores" in line:
-                break
-    return int(line.split()[-1])
+        lines = handler.readlines()
+
+    return len(list(filter(lambda x: "processor" in x, lines)))
```

Comment:
it does what it says, it counts how often processor appears in the file. ?

Reply:
yes, exactly.
  have to transform the filter which is an `iterator` into a `list` to be able to count the elements

Reply:
really beautiful.. :)